tmail-backend/mailbox/encrypted/api/src/main/scala/com/linagora/tmail/encrypted/EncryptedEmailContentStore.scala

```diff
@@ -0,0 +1,66 @@
+package com.linagora.tmail.encrypted
+
+import org.apache.james.blob.api.BlobId
+import org.apache.james.mailbox.model.MessageId
+import org.reactivestreams.Publisher
+import reactor.core.scala.publisher.SMono
+
+object EncryptedEmailContentStore {
+  val POSITION_NUMBER_START_AT: Int = 0
+}
+
+trait EncryptedEmailContentStore {
+  def store(messageId: MessageId, encryptedEmailContent: EncryptedEmailContent)
+
+  def delete(messageId: MessageId)
+
+  def retrieveFastView(messageId: MessageId): EncryptedEmailFastView
+
+  def retrieveDetailedView(messageId: MessageId): EncryptedEmailDetailedView
+
+  def retrieveAttachmentContent(messageId: MessageId, position: Int): BlobId
+
+  def storeReactive(messageId: MessageId, encryptedEmailContent: EncryptedEmailContent): Publisher[Unit] =
+    SMono.fromCallable(() => store(messageId, encryptedEmailContent))
+
+  def deleteReactive(messageId: MessageId): Publisher[Unit] =
+    SMono.fromCallable(() => delete(messageId))
+
+  def retrieveFastViewReactive(messageId: MessageId): Publisher[EncryptedEmailFastView] =
+    SMono.fromCallable(() => retrieveFastView(messageId))
+
+  def retrieveDetailedViewReactive(messageId: MessageId): Publisher[EncryptedEmailDetailedView] =
+    SMono.fromCallable(() => retrieveDetailedView(messageId))
+
+  def retrieveAttachmentContentReactive(messageId: MessageId, position: Int): Publisher[BlobId] =
+    SMono.fromCallable(() => retrieveAttachmentContent(messageId, position))
```

Comment:
I disagree with the way you implement reactive. You have some calls in your normal methods (like to store and retrieve attachments) making some block() calls, destroying the purpose of reactive. If you want reactive, I think you should then convert all your code reactive :)

Reply:
Yes, I know, I just want to default method in the interface class (Ex: less code for InMemory). We can override it when implementing (Cassandra here)